### PR TITLE
Bump black to stable version

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR bumps black to a stable version

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 def get_all_extras() -> Dict:
 
     cli_deps = [
-        "click>=7.0.0,<8.0.0",
+        "click==8.0.2",
```
Comment (angrybayblade) [inline setup.py:35]: Same as above

File: aea/test_tools/click_testing.py [hunk 2]
```diff
@@ -34,7 +34,6 @@
 import sys
 from typing import Optional
 
-from click._compat import string_types  # type: ignore
```
Comment (angrybayblade) [inline aea/test_tools/click_testing.py:37]: To support the click bump

File: aea/test_tools/click_testing.py [hunk 3]
```diff
@@ -58,7 +57,7 @@ def invoke(  # type: ignore
         exit_code = 0
 
         with self.isolation(input=input, env=env, color=color) as outstreams:
-            if isinstance(args, string_types):
+            if isinstance(args, str):
```
Comment (angrybayblade) [inline aea/test_tools/click_testing.py:60]: Same as above

File: aea/test_tools/click_testing.py [hunk 4]
```diff
@@ -106,4 +105,5 @@ def invoke(  # type: ignore
             exit_code=exit_code,
             exception=exception,
             exc_info=exc_info,
+            return_value=None,
```
Comment (angrybayblade) [inline aea/test_tools/click_testing.py:108]: Same as above

**Review by angrybayblade — COMMENTED** [review]
The issue was we were using a pre-release version of black which has some compatibility issues with python >= 3.9. When used outside of a tox env or directly in command line, the reason was two different click versions. In pipfile we had click version set to `8.0.4` and setup.py had `click>=7.0.0,<8.0.0` which means when we ran pipenv install it installed click `8.0.4` but just after that we also install open-aea which installed click `7.1.2`. This has no issues when we used python 3.7 or 3.8 but the pre-release version of black combined with click 7.1.2 had some issues with python 3.9 and 3.10. So I've bumped both of them to a compatible version

**Comment by angrybayblade** [comment]
You can ignore the python files that I have not reviewed they's just re-formatted by new black version.

File: tests/test_cli/test_push.py [hunk 5]
```diff
@@ -91,8 +91,11 @@ def test_vendor_ok(
         copy_tree_mock,
     ):
         """Test ok for vendor's item."""
-        with mock.patch("os.path.exists", side_effect=[False, True, False]):
-            self.invoke("push", "--local", "skill", "fetchai/echo")
+
+        with mock.patch("click.core._"):
+            with mock.patch("os.path.exists", side_effect=[False, True, False]):
+                self.invoke("push", "--local", "skill", "fetchai/echo")
```
Comment (angrybayblade) [inline tests/test_cli/test_push.py:97]: Just a patch to fix the test failure caused by new click version

File: setup.py (hunk 1 above)
Comment (DavidMinarsch) [reply setup.py:35]: Usually, this is strongly discouraged. Deps should be defined with wide ranges for maximum usability. But I gather that's not possible in this case?

**Review by DavidMinarsch — APPROVED** [review]

File: tests/test_cli/test_push.py (hunk 5 above)
Comment (DavidMinarsch) [reply tests/test_cli/test_push.py:97]: Does that hint at an issue with the implementation?

Comment (angrybayblade) [reply tests/test_cli/test_push.py:97]: No, this is just a case of `click.core._` method using `os.path.exists` method which is supposed to be patched with 3 side effects which caused the lack of effects crashing the test, So I patched the method.

**Pull request merged**